sql/src/main/java/io/crate/analyze/SetLicenseAnalyzedStatement.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import io.crate.expression.symbol.Symbol;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class SetLicenseAnalyzedStatement implements AnalyzedStatement {
+
+    static final String EXPIRY_DATE_TOKEN = "expirydate";
+    static final String ISSUED_TO_TOKEN = "issuedto";
+    static final String SIGN_TOKEN = "signature";
+
+    static final int LICENSE_TOKEN_NUM = 3;
+
+    static final Set<String> LICENSE_ALLOWED_TOKENS = new HashSet<>(
+        Arrays.asList(EXPIRY_DATE_TOKEN, ISSUED_TO_TOKEN, SIGN_TOKEN));
+
+    private final Symbol expirationDateSymbol;
+    private final Symbol issuedToSymbol;
+    private final Symbol signatureSymbol;
+
+
+    SetLicenseAnalyzedStatement(final Map<String, Symbol> valueSymbols) {
```

Comment:
I'd prefer it if we could be explicit about the required arguments here. So I'd change this to `SetLicenseAnalyzedStatement(Symbol expiryDate, Symbol issuedTo, Symbol signature)`

From a discoverability/usage perspective having a constructor that takes a `Map`, even though it actually requires a `Map` with specific keys is not nice. You've no clue what *should* be in the Map without reading the implementation.